src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -108,6 +109,26 @@ public async Task SendRequestBodyAsync(CancellationToken cancellationToken)
                 }
             }
 
+            public async Task SendRequestContentWithExpect100ContinueAsync(CancellationToken cancellationToken)
+            {
+                // Start timer and try to read response headers.
+                TaskCompletionSource<bool> allowExpect100ToContinue = new TaskCompletionSource<bool>();
+                var expect100Timer = new Timer(
+                            s => ((TaskCompletionSource<bool>)s).TrySetResult(true),
+                            allowExpect100ToContinue, _connection._pool.Settings._expect100ContinueTimeout, Timeout.InfiniteTimeSpan);
+                Task response = ReadResponseHeadersAsync(true, allowExpect100ToContinue);
+
+                // By now, either we got response from server or timer expired.
```

Comment:
nit: I think this would be better phrased as "Wait until we either get a response from server or timer expires."